clients/client-core/src/client/received_buffer.rs

```diff
@@ -438,13 +442,16 @@ impl FragmentedMessageReceiver {
         }
     }
 
-    async fn run_with_shutdown(&mut self, mut shutdown: task::TaskClient) {
+    async fn run_with_shutdown(
+        &mut self,
+        mut shutdown: task::TaskClient,
+    ) -> Result<(), MessageRecoveryError> {
         debug!("Started FragmentedMessageReceiver with graceful shutdown support");
         while !shutdown.is_shutdown() {
             tokio::select! {
                 new_messages = self.mixnet_packet_receiver.next() => {
                     if let Some(new_messages) = new_messages {
-                        self.received_buffer.handle_new_received(new_messages).await;
+                        self.received_buffer.handle_new_received(new_messages).await?;
```

Comment:
should we be returning an error and triggering shutdown if we received a malformed message?

Reply:
I'd rather return an error, but we should probably ticket that in a different PR, this will take a while...